pkg/docker/create.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/daytonaio/daytona/cmd/daytona/config"
+	"github.com/daytonaio/daytona/pkg/models"
+	"github.com/daytonaio/daytona/pkg/ports"
+	"github.com/daytonaio/daytona/pkg/ssh"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/go-connections/nat"
+	log "github.com/sirupsen/logrus"
+)
+
+func (d *DockerClient) CreateTarget(target *models.Target, targetDir string, logWriter io.Writer, sshClient *ssh.Client) error {
+	return nil
+}
+
+func (d *DockerClient) CreateWorkspace(opts *CreateWorkspaceOptions) error {
+	ctx := context.TODO()
+	mounts := []mount.Mount{}
+
+	var availablePort *uint16
+	portBindings := make(map[nat.Port][]nat.PortBinding)
+	portBindings["22/tcp"] = []nat.PortBinding{
```

Comment:
All the port bindings need to go to an available host port.

You can get an available port with `ports.GetAvailableEphemeralPort`